syntax = "proto3";

message SceneInfo
{
  uint32 scene_confid = 1;
  uint64 scene_id = 2;
}

message AfterChangeGsEnterScene
{
  SceneInfo scene_info = 1;
  uint32 gs_node_id = 2;
}

message MsChangeSceneInfo
{
  SceneInfo scene_info = 1;
  uint32 gs_node_id = 2;
  uint32 change_sequence = 3;
  bool change_gs_succeed = 4;//防止gs重启的时候gs_node_id一样
  bool change_gate_succeed = 5;//gate 切换完毕
  bool rg_change_succeed = 6;//gate 切换完毕
  uint64 change_time = 8;//设置超时，超时代表服务器崩溃了
  bool ignore_full = 9;//忽略人数满了
  bool is_leader_call = 10;//是否是队长叫换场景
  uint32 change_gs_type = 11;//0同服务器间切换,1不同服务器间切换
  uint32 cross_type = 12;//0不跨服,1跨服
}